readthedocs/organizations/forms.py

```diff
@@ -60,23 +60,24 @@ def __init__(self, *args, **kwargs):
             )
         super().__init__(*args, **kwargs)
 
-    def clean_name(self):
-        """Raise exception on duplicate organization slug."""
-        name = self.cleaned_data["name"]
+    def clean_slug(self):
+        slug_source = self.cleaned_data["slug"]
 
         # Skip slug validation on already created organizations.
         if self.instance.pk:
-            return name
+            return slug_source
 
-        potential_slug = slugify(name)
-        if not potential_slug:
-            raise forms.ValidationError(_("Invalid organization name: no slug generated"))
-        if Organization.objects.filter(slug=potential_slug).exists():
-            raise forms.ValidationError(
-                _("Organization %(name)s already exists"),
-                params={"name": name},
-            )
-        return name
+        slug = slugify(slug_source, dns_safe=True)
+        if not slug or slug != slug_source:
+            # There is a difference between the slug from the front end code, or
+            # the user is trying to submit the form without our front end code.
+            # TODO this should probably be individual validation rules, the
+            # reason for the validation error isn't going to be clear to the
+            # user.
+            raise forms.ValidationError(_("Invalid slug"))
```

Comment:
We can add the generated valid slug in the message, like "invalid slug, suggested slug: {slug}"